src/sysext/sysext.c

```diff
@@ -349,19 +458,40 @@ static int unmerge_hierarchy(
                 }
 
                 log_info("Unmerged '%s'.", p);
+                n_unmerged++;
         }
 
-        return 0;
+        return n_unmerged;
 }
 
 static int unmerge_subprocess(
                 ImageClass image_class,
-                char **hierarchies) {
+                char **hierarchies,
+                const char *workspace) {
 
         int r, ret = 0;
 
+        /* Mark the whole of /run as MS_SLAVE, so that we can mount stuff below it that doesn't show up on
+         * the host otherwise. */
+        r = mount_nofollow_verbose(LOG_ERR, NULL, "/run", NULL, MS_SLAVE|MS_REC, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to remount /run/ MS_SLAVE: %m");
+
+        /* Let's create the workspace if it's missing */
+        r = mkdir_p(workspace, 0700);
+        if (r < 0)
+                return log_error_errno(r, "Failed to create '%s': %m", workspace);
+
+        _cleanup_free_ char *stack_path = path_join(workspace, "submounts/stack");
```

Comment:
so, why do obstructed mounts matter to you? why bother with such "stacks" of mounts and replicating them 100% accurately? 

We can only replicate them if "peal" off the mounts bit by bit from the source tree, but that means this is never going to be atomic, and that sucks?

do you have a real-life scenario in mind with such stacks?

because to me the chance to make this somewhat atomic is much more interesting than to 100% being able to replicate the source tree. 99% suffice to me ;-)